src/client/routes/BUILD.bazel

```diff
@@ -1,9 +1,17 @@
-load("//bazel/js:defaults.bzl", "build_module", "build_route", "generate_route_manifest", "pkg_web")
+load("@npm//webpack-cli:index.bzl", "webpack_cli")
+load("//bazel/js:defaults.bzl", "build_module", "pkg_web")
+load("@rules_spa//spa:defs.bzl", "build_route", "generate_route_manifest")
 
 build_route(
     name = "default",
-    data = ["default.tsx"],
+    srcs = ["default.tsx"],
+    data = [
+        "//:package.json",
+        "@npm//:node_modules",
+    ],
     entry = "./$(execpath default.tsx)",
+    shared = "//bazel/js/internals/webpack:webpack.module-federation.shared.js",
+    webpack = webpack_cli,
```

Comment:
As long as the module federation config above fits a standard share pattern, then this doesn't need to be webpack if something else implements the pattern down the road. Maybe can we call this `build_tool` and default to `webpack_cli`?

Reply:
Basically, I think there is potential here to do what Alex did with the ts_project + SWC macro ahead of time. Though webpack + MF does have to completely overhaul the require system, that's something that either another tool will do or there will eventually come a native solution I think.

Reply:
Makes sense. Can we make an issue upstream and I can rename the parameter?

Reply:
Yes that sounds good!

Reply:
I will address this in another prerelease when I figure out how to switch based off of what tool was passed